pt_miniscreen/components/arrow_navigation_indicator.py

```diff
@@ -0,0 +1,87 @@
+import logging
+from enum import Enum, auto
+
+from PIL import ImageDraw
+
+from pt_miniscreen.core.component import Component
+from pt_miniscreen.core.utils import apply_layers, layer
+
+logger = logging.getLogger(__name__)
+
+
+class ArrowPosition(Enum):
+    TOP = auto()
+    BOTTOM = auto()
+
+
+class ArrowDrawing(Component):
```

Comment:
I'm not sure this needs to be a component, since the state is only in charge of position and whether to render or not which the parent can control already. Instead a "render function" similar to `rectangle` might make more sense:

```python3
def arrow(image, rotation=0, fill=1):
    center = (int(image.width / 2), int(image.height / 2))
    radius = int(min(image.width, image.height) / 2)
    ImageDraw.Draw(image).regular_polygon((center, radius), 3, fill=1, rotation=rotation)
    return image
```

then it can be used by `ArrowNavigationIndicator` like:
```python3
class ArrowNavigationIndicator(Component):
    def __init__(
        self,
        upper_arrow_padding=(0, 0),
        lower_arrow_padding=(0, 0),
        top_arrow_visible=True,
        bottom_arrow_visible=True,
        **kwargs
    ):
        super().__init__(
            **kwargs,
            initial_state={
                "upper_arrow_padding": upper_arrow_padding,
                "lower_arrow_padding": lower_arrow_padding,
                "top_arrow_visible": top_arrow_visible,
                "bottom_arrow_visible": bottom_arrow_visible
            },
        )

    def render(self, image):
        layers = []
        arrow_size = (10, 10)

        if self.state["upper_arrow_visible"]:
            layers.append(
                layer(
                    arrow,
                    size=arrow_size,
                    pos=self.state["upper_arrow_padding"],
                )
            )

        if self.state["bottom_arrow_visible"]:
            icon_padding = self.state["lower_arrow_padding"]
            icon_pos = (icon_padding[0], image.height - arrow_size[1] - icon_padding[1])

            layers.append(
                layer(partial(arrow, rotation=180), size=arrow_size, pos=icon_pos)
            )

        return apply_layers(image, layers)
```

I suspect there was also some 'hidden' padding because the arrow component size and the actual arrow size didn't match up. So yeah "arrow_size" might need to change for this to be the same as before.

Reply:
Side note: I wasn't really sure what to call these "render functions". I feel they need a name of some sort so that if they were in their own file they might be used like:

```python3
from pt_miniscreen.core.utils import render
...
class Test(Component):
    def render(self, image):
        return apply_layers(image, [
            layer(render.arrow), # <- or like render.rectangle or something.
        ])
```

maybe "paint" or "draw" is better? Not sure, would be interested in your opinion here!

```python3
from pt_miniscreen.core.utils import paint
...
class Test(Component):
    def render(self, image):
        return apply_layers(image, [
            layer(paint.arrow),
        ])
```

Reply:
`draw` is a good name ... I think using these functions as `draw.rectangle` or `draw.arrow` makes sense.... I also thought in `shapes` but `draw` is more comprehensive and would let us add more 'render functions' into that file later on

Reply:
yeah draw is tempting, my only worry is that it's too close to the terminology used by PIL for something that behaves differently to these functions. Anyhoo let's address it when the number of these utils get's a bit bigger! No need to overcommit to anything yet. :)